arroyo-df/src/physical.rs

```diff
@@ -156,7 +154,119 @@ pub fn window_scalar_function() -> ScalarUDF {
     )
 }
 
-impl FunctionRegistry for EmptyRegistry {
+#[derive(WrapperApi)]
+struct UdfDylibInterface {
+    run: unsafe extern "C" fn(
+        args: Vec<(FFI_ArrowArray, FFI_ArrowSchema)>,
```

Comment:
I don't think Vec is a safe interface for FFI (i.e., there's no guarantee that memory layout will remain the same across rust versions). Conveniently, Rust just published new docs on ABI compatibility: https://doc.rust-lang.org/stable/std/primitive.fn.html#abi-compatibility.